### PR TITLE
Encapsulate retargeting settings.

### PR DESCRIPTION
**Pull request by toxeus**

### CODE REVIEW

**Comment by evoskuil** [comment]
Repeating consensus code independently three times, and in a settings class, makes me uncomfortable. Please create utility functions to perform these operations. Static functions off of `chain_state` is probably best.

**Comment by toxeus** [comment]
Done.

File: include/bitcoin/bitcoin/chain/chain_state.hpp [hunk 1]
```diff
@@ -174,6 +174,13 @@ class BC_API chain_state
     /// This block height is less than or equal to that of the top checkpoint.
     bool is_under_checkpoint() const;
 
+    static uint32_t minimum_timespan(uint32_t retargeting_interval_seconds,
+        uint32_t retargeting_factor);
+    static uint32_t maximum_timespan(uint32_t retargeting_interval_seconds,
+        uint32_t retargeting_factor);
+    static uint32_t retargeting_interval(uint32_t retargeting_interval_seconds,
+        uint32_t block_spacing_seconds);
+
```
Comment (evoskuil) [inline include/bitcoin/bitcoin/chain/chain_state.hpp]: Style: list static functions at the top.

Comment (toxeus) [reply include/bitcoin/bitcoin/chain/chain_state.hpp]: Done.

**Pull request merged**